libs/filamat/src/MaterialBuilder.cpp

```diff
@@ -40,6 +40,9 @@
 #include "eiff/DictionaryTextChunk.h"
 #include "eiff/DictionarySpirvChunk.h"
 
+#define EMIT_CHUNK(chunkType, ...) ( container.addChild(std::make_unique<chunkType>(__VA_ARGS__)) )
```

Comment:
You could make a function with variadic templates instead of relying on the preprocessor.

Reply:
Done, added new methods to `ChunkContainer`.